lib/configure.js

```diff
@@ -39,10 +41,23 @@ function configure (gyp, argv, callback) {
       callback(err)
     } else {
       python = found
-      getNodeDir()
+      findVisualStudio2017()
     }
   })
 
+  function findVisualStudio2017() {
+    if (win) {
+      const vsSetup = findVS.getVS2017Setup();
+      if (!vsSetup) return callback()
+      gyp.opts.msvs_version = '2017';
+      process.env['vs2017_install'] = vsSetup.InstallationPath;
+      if (vsSetup.SDK) process.env['vs2017_sdk'] = vsSetup.SDK.replace(/\d+$/, '0')
+      callback();
+    } else {
+      getNodeDir()
+    }
+  }
+
```

Comment:
`getNodeDir()` always needs to run, and this function should only run if there was no version of VS specified, or it was 2017. Errors finding VS2017 should be ignored and it should continue to find other versions. What about:

```js
  findPython(python, function (err, found) {
    if (err) {
      callback(err)
    } else {
      python = found
      if (win)
        findVisualStudio2017()
      getNodeDir()
    }
  })

  function findVisualStudio2017() {
    if (gyp.opts.msvs_version && gyp.opts.msvs_version !== '2017')
      return

    try {
      const vsSetup = findVS.getVS2017Setup()
    } catch (_) {}
    if (!vsSetup)
      return

    gyp.opts.msvs_version = '2017'
    process.env['vs2017_install'] = vsSetup.InstallationPath
    if (vsSetup.SDK)
      process.env['vs2017_sdk'] = vsSetup.SDK.replace(/\d+$/, '0')
  }
```